src/OrleansServiceFabricUtils/FabricMembershipOracle.cs

```diff
@@ -366,25 +373,30 @@ private void ClearCaches()
             this.multiClusterSilosCache = null;
         }
 
-        private void NotifySubscribers(SiloAddress address, SiloStatus newStatus, List<ISiloStatusListener> listeners)
+        private async Task ProcessNotifications()
         {
-            Task.Factory.StartNew(() =>
+            while (!this.notifications.IsAddingCompleted)
             {
-                foreach (var subscriber in listeners)
+                await this.notificationEvent.WaitAsync(TimeSpan.FromMinutes(1));
```

Comment:
Infinite might work, but the timeout is benign and it's more 'obviously correct' to me. So this task will wake up every minute to explicitly check if either:
a) some notification needs processing, or
b) the silo has shutdown and so it can terminate